File: go/problems/rotate_array.go
package problems

// https://leetcode.com/problems/rotate-array/
func Rotate(nums []int, k int) []int {
	result := make([]int, len(nums))
	for index := range nums {
		result[index] = nums[(index+k)%len(nums)]
	}
	return result
}

File: rust/src/problems/rotate_array.rs
// https://leetcode.com/problems/rotate-array/
pub fn rotate(nums: Vec<i32>, k: i32) -> Vec<i32> {
  return nums
    .iter()
    .enumerate()
    .map(|(i, _)| return nums[(i + k as usize) % nums.len()])
    .collect();
}
